modules/siddhi-extensions/math/src/main/java/org/wso2/siddhi/extension/math/PercentileFunctionExtension.java

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.extension.math;
+
+import org.wso2.siddhi.core.config.ExecutionPlanContext;
+import org.wso2.siddhi.core.exception.ExecutionPlanCreationException;
+import org.wso2.siddhi.core.exception.OperationNotSupportedException;
+import org.wso2.siddhi.core.executor.ConstantExpressionExecutor;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.core.query.selector.attribute.aggregator.AttributeAggregator;
+
+import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * percentile(values, p)
+ * Returns an estimate for the pth percentile of the stored values
+ * Accept Type(s): FLOAT,INT,LONG,DOUBLE
+ * Return Type: DOUBLE
+ */
+public class PercentileFunctionExtension extends AttributeAggregator {
+
+    private PercentileFunctionExtension percentileOutputFunctionExtension;
+    private double percentileValue;
+
+    /**
+     * The initialization method for FunctionExecutor
+     *
+     * @param attributeExpressionExecutors are the executors of each attributes in the function
+     * @param executionPlanContext Execution plan runtime context
+     */
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ExecutionPlanContext executionPlanContext) {
+
+        if (attributeExpressionExecutors.length != 2) {
+            throw new OperationNotSupportedException("Percentile function has to have exactly 2 parameter, currently "
+                    + attributeExpressionExecutors.length + " parameters provided");
+        }
+
+        try {
+
+            if (attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor) {
+                percentileValue = ((Double) attributeExpressionExecutors[1].execute(null));
+            } else {
+                throw new OperationNotSupportedException("Percentile value has to be a constant");
+            }
+
+        } catch (ClassCastException c) {
```

Comment:
Yes, I can understand your point now and It's true. Surely it will improve the readability if we use instanceof check rather than catching a ClassCastException. I checked above links also and seems like there is not much performance difference also. As you said since this is the init() method surely the best option would be using an instanceof check. Thanks @dilini-muthumala and @lasanthafdo for your feedbacks. I will do the necessary changes.